code-challenges.test.js

```diff
@@ -23,9 +42,56 @@ const hitchhikersCharacters = [
 
 // b) Create the function that makes the test pass.
 
-// --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+  // Pseudo coding:
+  // input:arr
+  // output:returns an array with a sentence about each person with their name capitalized.
+  // process: create a function name: pplJob
+  // create a new variable newArr = []
+  // use for loop starting at 0, ending is smaller then arr.length iteration is i++
+  // then make new variable capFirst:
+      //  use .split(' ') with a space on arr.name to sperate first and last name and use [0] index to get first name and mention [0] index again ref first letter in frist name and use .charAt with 0 index then use .toUppercase() to capitalize the letter 
+      // then adding remaining word by using .slice on first name to remove the first letter
+  // create a new variable capLast:
+      //  using same logic as first name but mention [1] index to get last name then mention [0] again to get first letter and capitalize it
+      // then adding remaining word of last name using same logic with .slice to remove first letter
+  // then return using .push on newArr with capFirst + capLast and arr[i].occupation in a sentence.
+  // pass test
 
-// a) Create a test with an expect statement using the variables provided.
+    // const pplJob = (arr) => {
+    //   const newArr = []
+    //   for (let i = 0; i < arr.length; i++) {
+    //     const capFirst = arr[i].name.split(' ')[0][0].toUpperCase() + arr[i].name.split(' ')[0].slice(1)
+    //     const capLast = arr[i].name.split(' ')[1][0].toUpperCase() + arr[i].name.split(' ')[1].slice(1)
+    //     newArr.push(`${capFirst} ${capLast} is ${arr[i].occupation}.`)
```

Comment:
Interesting approach here. I suggest refactoring so that it will just capitalize every first letter of every word in the `name` key, instead of only looking for index 0 and 1. Also, instead of making an empty array to push things into, what HOF can you use that will do that for you?